statsd/statsdtest/recorder.go

```diff
@@ -0,0 +1,85 @@
+package recorder
+
+import (
+	"errors"
+	"sync"
+
+	"github.com/falun/go-statsd-client/statsd"
+)
+
+// RecordingSender implements statsd.Sender but parses individual Stats into a
+// buffer that can be later inspected instead of sending to some server. It
+// should constructed with NewRecordingSender().
+type RecordingSender struct {
+	m      sync.Mutex
+	buffer Stats
+	closed bool
+}
+
+var _ statsd.Sender = &RecordingSender{}
```

Comment:
Is this being used as a compilation level type enforcement, where if it doesn't match interface, this will not compile?
If so, it may make more sense for this to be in the associated test file... or to have a test case specifically test that a type assertion is "ok".